Floating Point Error Estimation RST added

QuillPusher opened the pull request:
- Floating Point Error Estimation Info and links consolidated in this article

- Added link to Side Navigation in Index file (shortened the name to "FP" due to limited space in side menu)

vgvassilev commented on docs/userDocs/source/user/FloatingPointErrorEstimation.rst:
```diff
@@ -0,0 +1,199 @@
+floating-point error detection can help reduce data reprocessing costs and help
+develop important new lossy compression algorithms.
+
+Who is this article for?
```
Perhaps we can remove this section?

QuillPusher replied:
section removed and commit squashed to existing commit

parth-07 commented on docs/userDocs/source/user/FloatingPointErrorEstimation.rst:
```diff
@@ -0,0 +1,191 @@
+
+AD helps evaluate the exact derivative of a function. AD applies the chain rule
+of differential calculus throughout the semantics of the original program. In
+the context of FP error estimation, this research uses:
```
Can you please clarify what do you mean by 'this research uses'?

QuillPusher replied:
Thanks, I just realized that the phrasing is a bit off... Rephrasing as suggested by Vassil below:

AD helps evaluate the exact derivative of a function. AD applies the
**differential calculus chain rule** throughout the semantics of the original
program. In the context of FP error estimation, **~this research~ the implementation relies on:**:

parth-07 commented on docs/userDocs/source/user/FloatingPointErrorEstimation.rst:
```diff
@@ -0,0 +1,191 @@
+Where does the FPEE logic reside?
+=================================
+
+The main logic for Floating Point Error Estimation resides in the following
```
We should not add implementation details in the user documentation. For user documentation, we should perhaps focus on these things:

- What is error estimation? 
- Why use AD in error estimation as opposed to other differentiation techniques in error estimation?
- What will happen if AD fails?
- Tutorials/Examples that show how to use Clad's error estimation functionality effectively.

@vgvassilev and @grimmmyshini can comment better on the documentation structure.

QuillPusher replied:
I have linked the Doxygen and ReadMe docs in the comment below

parth-07 commented on docs/userDocs/source/user/FloatingPointErrorEstimation.rst:
```diff
@@ -0,0 +1,191 @@
+
+While parsing the code using Clad, if it encounters a floating point
+variable,it needs to be registered into the system (to accumulate relevant
+errors against that variable). Next, the Error Estimation Calcualtion Formula
```
```suggestion
errors against that variable). Next, the Error Estimation Calculation Formula
```

parth-07 commented on docs/userDocs/source/user/FloatingPointErrorEstimation.rst:
```diff
@@ -0,0 +1,191 @@
+=============================
+
+While parsing the code using Clad, if it encounters a floating point
+variable,it needs to be registered into the system (to accumulate relevant
```
What is system here?

QuillPusher replied:
Rephrased as suggested by Vassil:

While parsing the code using Clad, if it encounters a floating point variable,
it needs to be **tracked**

parth-07 commented on docs/userDocs/source/user/FloatingPointErrorEstimation.rst:
```diff
@@ -0,0 +1,191 @@
+  For this formula to work, the value of the variable (delta_x) should be saved
+  at the relevant time.
+
+This model will return a formula that is represented using a Clang
```
Users perhaps do not care at all if the error is a clang expression. We should write this documentation in a way that is useful for the users.

QuillPusher replied:
This is as described by Garima. As I understand, the idea was that familiarity with Clang expressions is required, since we're targeting a dev audience, including new comers... Please let me know if you have an alternate text suggestion

vgvassilev commented on docs/userDocs/source/user/FloatingPointErrorEstimation.rst:
```diff
@@ -0,0 +1,191 @@
+=============================
+
+While parsing the code using Clad, if it encounters a floating point
+variable,it needs to be registered into the system (to accumulate relevant
```
```suggestion
variable, it needs to be tracked (to accumulate relevant
```

QuillPusher commented:
Thanks @parth-07  and @vgvassilev for the review, requested changes incorporated

vgvassilev commented:
@parth-07, can you take a look again?

@davidlange6, could you take a look and proofread?

davidlange6 commented on docs/userDocs/source/user/FloatingPointErrorEstimation.rst:
```diff
@@ -0,0 +1,196 @@
+FP Error Estimation
```
Don't abbreviate the title.. Write out Floating Point

davidlange6 replied:
Branding - this tool is called CHEF-FP. We should refer to it that way (repeatedly)

davidlange6 commented on docs/userDocs/source/user/FloatingPointErrorEstimation.rst:
```diff
@@ -0,0 +1,196 @@
+
+.. _demos/ErrorEstimation: https://github.com/vgvassilev/clad/tree/master/demos/ErrorEstimation
+
+.. _Fast and Automatic Floating Point Error Analysis: https://arxiv.org/pdf/2304.06441.pdf
```
use complete title

QuillPusher commented:
Incorporated David's review changes:
- Mentioned Chef-FP in relevant places
- Explained Chef-FP in the beginning, linked to paper
- changed FPEE to full form in all places (except in code elements)
- squashed commits for a cleaner look
